Improve `Mesh` class description

mateuseap opened the pull request:
## What I did 

- Bind `MAX_MESH_SURFACES` constant in `RenderingServer::_bind_methods()`
- Add `MAX_MESH_SURFACES` constant to `RenderingServer` class documentation
- Reference the constant `RenderingServer.MAX_MESH_SURFACES` in the description of `Mesh` class documentation

Closes: https://github.com/godotengine/godot-docs/issues/8720

mateuseap commented:
@timothyqiu I have bound the constant `MAX_MESH_SURFACES` in the `RenderingServer::_bind_methods()` method. Additionally, I have added documentation for this constant in the `RenderingServer` class and referenced it in the `Mesh` class documentation, instead of leaving the value 256 hard-coded in the description.

timothyqiu commented on doc/classes/RenderingServer.xml:
```diff
@@ -4239,6 +4239,9 @@
 			The maximum number of directional lights that can be rendered at a given time in 2D.
 		</constant>
+		<constant name="MAX_MESH_SURFACES" value="256">
+			Maximum number of surfaces a [Mesh] can have.
```
```suggestion
			The maximum number of surfaces a mesh can have.
```

- Keep the format the same as other `MAX_*` constants.
- The `Mesh` resource is built on top of `RenderingServer`. It's just a general concept "mesh" here (created by `mesh_create()`).

mateuseap replied:
Done! Thanks for the adjust suggestion

mateuseap committed:
Improve Mesh class description

- Bind MAX_MESH_SURFACES constant in RenderingServer::_bind_methods()
- Add MAX_MESH_SURFACES constant to RenderingServer class documentation
- Reference the constant RenderingServer.MAX_MESH_SURFACES in the description of Mesh class documentation
